model/itemTab.cc

```diff
@@ -458,19 +459,21 @@ namespace minsky
                     if (it==itemFocus) {
                       cairo_translate(cairo,xItem,yItem);  		    				   
                       itemCoords.erase(itemFocus);   
-                      itemCoords.emplace(make_pair(itemFocus,make_pair(xItem,yItem)));         
+                      itemCoords.emplace(make_pair(itemFocus,make_pair(xItem,yItem)));
                     } else cairo_translate(cairo,itemCoords[it].first,itemCoords[it].second);      
                     p->draw(cairo);
                   }
-                else if (auto g=dynamic_cast<GodleyIcon*>(it.get()))
+                else if (auto g=dynamic_pointer_cast<GodleyIcon>(it))
                   {
                     cairo::CairoSave cs(cairo);   
                     if (it==itemFocus) {
                       cairo_translate(cairo,xItem,yItem);  		    				   
                       itemCoords.erase(itemFocus);   
                       itemCoords.emplace(make_pair(itemFocus,make_pair(xItem,yItem)));         
-                    } else cairo_translate(cairo,itemCoords[it].first,itemCoords[it].second);      
-                    g->draw(cairo);
+                    } else cairo_translate(cairo,itemCoords[it].first,itemCoords[it].second);
+                    auto godley=GodleyTableWindow(g);
```

Comment:
Cleaner would be GodleyTableWindow godley(g); The above may invoke an unnecessary copy constructor.